src/main/java/com/github/redouane59/twitter/helpers/RequestHelperV2.java

```diff
@@ -45,14 +46,14 @@ public RequestHelperV2(TwitterCredentials twitterCredentials, OAuth10aService se
     return makeRequest(Verb.GET, url, parameters, null, true, classType);
   }
   
-  public void getAsyncRequest(String url, Consumer<Tweet> consumer) {
-	  getAsyncRequest(url, consumer, TweetV2.class);
+  public Future<Response> getAsyncRequest(String url, Consumer<Tweet> consumer) {
+	  return getAsyncRequest(url, consumer, TweetV2.class);
   }
 
-  public <T> void getAsyncRequest(String url, final Consumer<T> consumer, final Class<? extends T> targetClass) {
+  public <T> Future<Response> getAsyncRequest(String url, final Consumer<T> consumer, final Class<? extends T> targetClass) {
```

Comment:
Can you quickly explain me what this is bringing ? Thanks !

Reply:
`Future` gives at least some control over that spawned parallel thread. User can cancel or by invoking `get` join that thread.

Reply:
Awesome !